subdoc/lib/clang_resource_dir.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "subdoc/lib/clang_resource_dir.h"
+
+#include <filesystem>
+#include <fstream>
+
+#include "subdoc/llvm.h"
+
+namespace subdoc {
+
+Option<std::string> ClangResourceDir::find_resource_dir(std::string_view tool) {
+  if (auto it = cache_.find(tool); it != cache_.end()) {
+    return sus::some(it->second);
+  }
+
+  std::filesystem::path tool_path(tool);
+  std::string stem = tool_path.stem().string();
+  if (!stem.starts_with("clang")) return sus::none();
+
+  if (!std::filesystem::exists(tool_path)) {
+    fmt::println(stderr, "WARNING: can't find clang compiler at '{}'", tool);
+    return sus::none();
+  }
+
+  std::array<llvm::StringRef, 2> args = {tool, "-print-resource-dir"};
+  if (stem.find("clang-cl") != std::string::npos) {
```

Comment:
Shouldn't this be a .startswith()? Granted if you name your compiler `clang-clang-cl-18` you deserve everything bad that will happen to you, but still...

Reply:
Yeah fair, it can be.